cmd/argocd/commands/repo.go

```diff
@@ -42,17 +41,25 @@ func NewRepoAddCommand(clientOpts *argocdclient.ClientOptions) *cobra.Command {
 		repo                  appsv1.Repository
 		upsert                bool
 		sshPrivateKeyPath     string
+		caFile                string
+		certFile              string
+		keyFile               string
+		repoType              string
 		insecureIgnoreHostKey bool
 	)
 	var command = &cobra.Command{
 		Use:   "add REPO",
-		Short: "Add git repository credentials",
+		Short: "Add repository credentials",
 		Run: func(c *cobra.Command, args []string) {
 			if len(args) != 1 {
 				c.HelpFunc()(c, args)
 				os.Exit(1)
 			}
 			repo.Repo = args[0]
+			repo.Type = appsv1.RepoType(repoType)
```

Comment:
This changes the behavior of this command. You MUST specify `--username`,  if you do not specify a password, then it will request it from you.